src/KafkaClient/KafkaOptions.cs

```diff
@@ -49,6 +52,11 @@ public class KafkaOptions
         public IConnectionConfiguration ConnectionConfiguration { get; }
 
         /// <summary>
+        /// Configuration for SSL encrypted communication
+        /// </summary>
+        public ISslConfiguration SslConfiguration { get; }
```

Comment:
Since ssl is part of the connection, what about this being a property of the IConnectionConfiguration? That should also help to reduce the number of places we need to pass this in.

Can you think of a reason we'd want to keep them separate, or change them independently?